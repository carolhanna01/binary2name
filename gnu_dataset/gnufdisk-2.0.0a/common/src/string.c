/* GNU Fidsk (gnufdisk-common), a library of common functions.
 *
 * Copyright (C) 2011 Free Software Foundation
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place - Suite 330,
 * Boston, MA 02111-1307, USA. */

#include <string.h>
#include <stdlib.h>
#include <stdarg.h>
#include <errno.h>

#include <gnufdisk-common.h>

struct gnufdisk_string {
  char* data;
  size_t size;
};

static int check_string(struct gnufdisk_string* _s)
{
  if(gnufdisk_check_memory(_s, sizeof(struct gnufdisk_string), 0) != 0
     || (_s->data != NULL && gnufdisk_check_memory(_s->data, _s->size, 0) != 0))
    return EFAULT;
  return 0;
}

struct gnufdisk_string* gnufdisk_string_new(const char* _fmt, ...)
{
  va_list args;
  struct gnufdisk_string* ret;
  int err;

  if((ret = malloc(sizeof(struct gnufdisk_string))) == NULL)
    {
      err = errno;
      goto lb_out;
    }

  memset(ret, 0, sizeof(struct gnufdisk_string));
    
  va_start(args, _fmt);
  ret->size = gnufdisk_vasprintf(&ret->data, _fmt, args);
  va_end(args);

  if(ret->size < 0)
    {
      err = errno;
      
      free(ret);
      ret = NULL;
    }

lb_out:
  
  errno = err;
  return ret;
}

int gnufdisk_string_set(struct gnufdisk_string* _p, const char* _fmt, ...)
{
  va_list args;
  char* s;
  int err;
  int ret;

  if((err = check_string(_p)) != 0)
    {
      ret = -1;
      goto lb_out;
    }
  
  va_start(args, _fmt);
  ret = gnufdisk_vasprintf(&s, _fmt, args);
  va_end(args);

  if(ret < 0)
    {
      ret = -1;
      err = errno;
      goto lb_out;
    }
  else 
    err = 0;

  if(_p->data)
    free(_p->data);

  _p->data = s;
  _p->size = ret;

lb_out:
  
  errno = err;
  return ret;
}

int gnufdisk_string_length(struct gnufdisk_string* _p)
{
  if((errno = check_string(_p)) != 0)
    return -1;

  return strlen(_p->data != NULL ? _p->data : "");
}

const char* gnufdisk_string_c_string(struct gnufdisk_string* _p)
{
  if((errno = check_string(_p)) != 0)
    return NULL;

  return _p->data;
}

char* gnufdisk_string_c_string_dup(struct gnufdisk_string* _p)
{
  char *r;

  if((errno = check_string(_p)) != 0)
    return NULL;

  r = strdup(_p->data ? _p->data : "");

  if(r == NULL)
    errno = ENOMEM;

  return r;
}

int gnufdisk_string_delete(struct gnufdisk_string* _s)
{
  if((errno = check_string(_s)) != 0)
    return -1;

  if(_s->data)
    free(_s->data);

  free(_s);

  return 0;
}

