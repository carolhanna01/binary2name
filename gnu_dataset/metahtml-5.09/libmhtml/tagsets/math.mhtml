;;; math.mhtml: -*- Meta-HTML -*-  A few math helpers.
;;; 
;;;  Copyright (c) 1998 Brian J. Fox
;;;  Author: Brian J. Fox (bfox@ai.mit.edu) Sun Apr 19 13:21:39 1998.
<defsubst float whitespace=delete>
  ;;; DOC_SECTION (ARITHMETIC-OPERATORS)
  ;;; Returns the floating point representation of <var x>.
  ;;; <var x> may be an integer, a floating point number, or a variable
  ;;; containing an integer or floating point number.
  ;;; <complete-example>
  ;;; <set-var foo=7>
  ;;; <float foo>
  ;;; <float 3>
  ;;; <float 3.0>
  ;;; </complete-example>
  <mul 1.0 %0>
</defsubst>

<defun number? arg &key base>
  ;;; DOC_SECTION (ARITHMETIC-OPERATORS)
  ;;; Returns "true" if <var arg> is the string representation of an integer
  ;;; in base <var base> (default <code>10</code>), or the string
  ;;; representation of a floating point number in base 10.
  ;;;
  ;;; For integer checks, the special value of zero (<code>0</code>) for
  ;;; <var base> allows the common radixes of decimal, octal, and hexadecimal
  ;;; to be understood.  That is to say:
  ;;; <example>
  ;;; <number? <get-var x> base=0>
  ;;; </example>
  ;;; is equivalent to:
  ;;; <example>
  ;;; <or <integer? <get-var x> base=8>
  ;;;     <integer? <get-var x> base=10>
  ;;;     <integer? <get-var x> base=16>
  ;;;     <real?    <get-var x>>>
  ;;; </example>
  ;;;
  ;;; Some examples:
  ;;; <example>
  ;;; <number? 10>          --> true
  ;;; <number? .9>          --> true
  ;;; <number 0xEF base=16> --> true
  ;;; </example>
  <defvar base 10>
  <or <integer? <get-var arg> base=<get-var base>> <real? <get-var arg>>>
</defun>

<defun avg &rest nums[]>
  ;;; DOC_SECTION (ARITHMETIC-OPERATORS)
  ;;; Returns the average of all of the arguments passed.
  ;;; Examples:
  ;;; <example>
  ;;; <avg 3 4 5>    --> 4
  ;;; <avg 4 9 3.2>  --> 5.40
  ;;; </example>
  <set-var divisor = <array-size nums>>
  <set-var dividend = <%%backquote-eval <add ,<get-var nums[]>>>>
  <div dividend divisor>
</defun>

<defun factorial x>
  ;;; DOC_SECTIONS (ARITHMETIC-OPERATORS)
  ;;; Computes the factorial of <var x>.
  ;;; Example:
  ;;; <complete-example>
  ;;; <factorial 12>
  ;;; </complete-example>
  <if <lt x 2>
      1
    <mul <float x> <fact <sub x 1>>>>
</defun>
