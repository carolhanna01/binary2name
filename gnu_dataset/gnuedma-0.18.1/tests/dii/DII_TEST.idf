[General]
Name=DII_TEST
[SYS_DEF]
ModuleName=libDII_TEST.so
NameSpace=examples
MajorVer=0
MinorVer=0
[Definition]
PropertiesNum=2
MethodsNum=8
[Prop0]
Name=num
Type=ESINT32
Access=READ/WRITE
ArrayElems=
[Prop1]
Name=str
Type=EZSTRING
Access=READ/WRITE
ArrayElems=
[Met0]
Name=test_in_int
Signature=S32U32S16U16S8U8rS32
Virtual=0
[Met1]
Name=test_in_real
Signature=R32R64rS32
Virtual=0
[Met2]
Name=test_in_mixed
Signature=S8U32R32S32S32
Virtual=0
[Met3]
Name=test_out_int
Signature=sS8sU32sS32
Virtual=0
[Met4]
Name=test_out_real
Signature=sR32sR64rS32
Virtual=0
[Met5]
Name=test_in_other
Signature=ZOZrS32
Virtual=0
[Met6]
Name=test_out_other
Signature=sZsOsZrS32
Virtual=0
[Met7]
Name=test_inout
Signature=sR64ZsZS32sA
Virtual=0
[]
