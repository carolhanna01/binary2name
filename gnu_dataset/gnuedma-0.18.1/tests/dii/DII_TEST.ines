ClassName=DII_TEST
NameSpace=dii
Machine=i386
OperatingSystem=LINUX
Implementation=libDII_TEST
IDFParser=EDMAIDF
MajorVer=0
MinorVer=0
