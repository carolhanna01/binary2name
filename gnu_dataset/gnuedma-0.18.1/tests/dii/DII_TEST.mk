#   EDMA: Entorno de Desarrollo Modular y Abierto
#   Object Oriented and Componentware Framework  
#
#   Copyright (C) 1998, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2010
#             by David Mart√≠nez Oliveira
#   
#   This program is free software: you can redistribute it and/or modify
#   it under the terms of the GNU General Public License as published by
#   the Free Software Foundation, either version 3 of the License, or
#   (at your option) any later version.
#
#   This program is distributed in the hope that it will be useful,
#   but WITHOUT ANY WARRANTY; without even the implied warranty of
#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#   GNU General Public License for more details.
#   
#   You should have received a copy of the GNU General Public License
#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
#
# -----------------------------------------------------
# Makefile for EDMA class DII_TEST
# Generated by EDMA IDF Wizard Gnomized 0.1
#
CC=gcc
CCFLAGS=`edma-config --cflags-class`
LIBS=`edma-config --libs-class` 
NAMESPACE=examples

libDII_TEST.so: temp.o 
	$(CC) -nostartfiles -shared -Wl,-soname,libDII_TEST.so.1 -o $@ temp.o $(LIBS) 

temp.o: DII_TEST_imp.c 
	$(CC)  $(CCFLAGS) $< -o $@ 


.PHONY: install
install:
	mkdir -p `edma-config --dir-impl`/${NAMESPACE} 
	mkdir -p `edma-config --dir-int`/${NAMESPACE}
	cp libDII_TEST.so `edma-config --dir-impl`/edma/${NAMESPACE}/libDII_TEST.so.0.0.0
	cp DII_TEST.idf `edma-config --dir-int`/${NAMESPACE}

.PHONY: clean
clean:
	rm -f libDII_TEST.so *~ *.o core

