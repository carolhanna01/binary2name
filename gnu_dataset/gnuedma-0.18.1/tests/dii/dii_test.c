/*
 * EDMA: Entorno de Desarrollo Modular y Abierto
 * Object Oriented and Componetware Framework
 * Copyright (C) 1998, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2010
 *    David Martínez Oliveira
 *
 * This file is part of EDMA.
 *
 * EDMA is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * EDMA is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with EDMA.  If not, see <http://www.gnu.org/licenses/>.
 */

#include <stdio.h>
#include <string.h>
#include <math.h>
#include <edma.h>

int
main(int argc,char *argv[])
{
  OBJID            id, id_dii;
  ESint32          sin32;
  EUint32          uin32;
  ESint8           sin8;
  EReal32          re32;
  EReal64          re64;
  EChar            str1[80], str2[80];
  //EDMAT_BUFFER     in_buf;
  //EDMAT_BUFFER     out_buf;
  OBJID            obj;

  /* Initialize EDMA System*/
  EDMAInit();
  /* Create object */
  id = edma_new_obj ("DII_TEST");
  id_dii = edma_new_obj ("EDMA_DII");

  /* Test integer input parameters */
  edma_met3 (id_dii, "request", id, "test_in_int", "S32U32S16U16S8U8rS32");
  edma_met3 (id_dii, "add_sint32_param", -100000, 0);
  edma_met3 (id_dii, "add_uint32_param", 100000, 0);

  edma_met3 (id_dii, "add_sint16_param", -10000, 0);
  edma_met3 (id_dii, "add_uint16_param", 10000, 0);

  edma_met3 (id_dii, "add_sint8_param", -100, 0);
  edma_met3 (id_dii, "add_uint8_param", 100, 0);

  edma_printf ("%s", "=================> Running Method");
  edma_met3 (id_dii, "invoke", "");
  edma_printf ("%s", "=================> Method Finished =========================>");

  /* Test Real input parameters */
  edma_met3 (id_dii, "request", id, "test_in_real", "R32R64rS32");
  edma_met3 (id_dii, "add_r32_param", 5.123, 0);
  edma_met3 (id_dii, "add_r64_param", -500.321, 0); 

  edma_printf ("%s", "=================> Running Method");
  edma_met3 (id_dii, "invoke", "");
  edma_printf ("%s", "=================> Method Finished =========================>");

  /* Test Integer/Real input parameters */
  edma_met3 (id_dii, "request", id, "test_in_mixed", "S8U32R32S32S32");
  edma_met3 (id_dii, "add_sint8_param", 5, 0);
  edma_met3 (id_dii, "add_uint32_param", 10000, 0); 
  edma_met3 (id_dii, "add_r32_param", 100.523, 0); 
  edma_met3 (id_dii, "add_sint32_param", -10000, 0); 
  
  edma_printf ("%s", "=================> Running Method");
  edma_met3 (id_dii, "invoke", "");
  edma_printf ("%s", "=================> Method Finished =========================>");

  /* Test other types input parameters*/
  edma_met3 (id_dii, "request", id, "test_in_other", "ZOZrS32");
  edma_met3 (id_dii, "add_strz_param", "Hello World", 0);
  edma_met3 (id_dii, "add_obj_param", id, 0); 
  edma_met3 (id_dii, "add_strz_param", "bye cruel World!", 0); 
  
  edma_printf ("%s", "=================> Running Method");
  edma_met3 (id_dii, "invoke", "");
  edma_printf ("%s", "=================> Method Finished =========================>");

  /* Test Integer output parameters */

  edma_met3 (id_dii, "request", id, "test_out_int", "sS8sU32sS32");

  /* Note: Current EDMA_DII class do not process signature modifer 's' for output parameters
   * It also do not support output parameter in the sense that an output should be a pointer
   * A pointer is a 32-bit value (for x86 machines) that do not fits on a byte (first param)*/
  //edma_met3 (id_dii, "request", id, "test_out_int", "S32U32S32");

  sin8 = 100;
  uin32 = 10000;
  sin32 = -10000;

  /*
  //edma_met3 (id_dii, "add_sint8_param", &sin8, 0);
  edma_met3 (id_dii, "add_sint32_param", &sin8, 0); 
  edma_met3 (id_dii, "add_uint32_param", &uin32, 0);  
  edma_met3 (id_dii, "add_sint32_param", &sin32, 0); 
  */
  edma_met3 (id_dii, "add_sint8_out_param", &sin8, 0);
  edma_met3 (id_dii, "add_uint32_out_param", &uin32, 0); 
  edma_met3 (id_dii, "add_sint32_out_param", &sin32, 0);

  edma_printf ("%s", "=================> Running Method");
  edma_met3 (id_dii, "invoke", "");
  edma_printf ("%s", "=================> Method Finished =========================>");
  edma_printf ("%s", "Output parameters are:");
  edma_printf ("--> Signed 8-bit Integer    : %d [%p]", sin8, &sin8);
  edma_printf ("--> Unsigned 32-bit Integer : %d [%p]", uin32, &uin32);
  edma_printf ("--> Signed 32-bit Integer   : %d [%p]", sin32, &sin32);

  /* Test Real output parameters */
  edma_met3 (id_dii, "request", id, "test_out_real", "sR32sR64rS32");
  re32 = 10.56;
  re64 = 501.76;
    
  edma_met3 (id_dii, "add_r32_out_param", &re32, 0);
  edma_met3 (id_dii, "add_r64_out_param", &re64, 0);

  edma_printf ("%s", "=================> Running Method");
  edma_met3 (id_dii, "invoke", "");
  edma_printf ("%s", "=================> Method Finished =========================>");
  edma_printf ("%s", "Output parameters are:");
  edma_printf ("--> Real 32-bit Integer : %lf [%p]", re32, &sin8);
  edma_printf ("--> Real 64-bit Integer : %lf [%p]", re64, &uin32);

  /* Test other input */
  edma_met3 (id_dii, "request", id, "test_in_other", "ZOZrS32");
    
  edma_met3 (id_dii, "add_strz_param", "Hello", 0);
  edma_met3 (id_dii, "add_obj_param", id, 0);
  edma_met3 (id_dii, "add_strz_param", "World", 0);

  edma_printf ("%s", "=================> Running Method");
  edma_met3 (id_dii, "invoke", "");

  /* Test other output */
  edma_met3 (id_dii, "request", id, "test_out_other", "sZsOsZrS32");

  strcpy (str1, "Hello");
  strcpy (str2, "World!");
  obj = id_dii;

  edma_met3 (id_dii, "add_strz_out_param", str1, 0);
  edma_met3 (id_dii, "add_obj_out_param", &obj, 0);
  edma_met3 (id_dii, "add_strz_out_param", str2, 0);

  edma_printf ("%s", "=================> Running Method");
  edma_met3 (id_dii, "invoke", "");
  edma_printf ("%s", "=================> Method Finished =========================>");
  edma_printf ("%s", "Output parameters are:");
  edma_printf ("---> String 1 : '%s' ", str1);
  edma_printf ("---> Object   : %d", obj);
  edma_printf ("---> String 2 : '%s'", str2);

  /* Test input output */
  edma_met3 (id_dii, "request", id, "test_inout", "sR64ZsZS32sA");
  
  re64 = 101.79;
  strcpy (str1, "World!!!!");

  edma_met3 (id_dii, "add_r64_out_param", &re64, 0);
  edma_met3 (id_dii, "add_strz_param", "Hello", 0);
  edma_met3 (id_dii, "add_strz_out_param", str1, 0);
  edma_met3 (id_dii, "add_sint32_param", 54, 0);
  edma_met3 (id_dii, "add_sint32_param", 100, 0);

  edma_printf ("%s", "=================> Running Method");
  edma_met3 (id_dii, "invoke", "");
  edma_printf ("%s", "=================> Method Finished =========================>");
  edma_printf ("%s", "Output parameters are:");
  edma_printf ("---> String 1 : '%s' ", str1);
  edma_printf ("---> Real 64  : %lf", re64);

  /* Shutdown EDMA Suystem */
  edma_free_obj (id);
  edma_free_obj (id_dii);
  EDMAEnd();
  return 0;
}
