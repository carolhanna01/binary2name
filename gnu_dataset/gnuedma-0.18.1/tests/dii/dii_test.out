
GNU/EDMA 0.16.4 build [Jan 30 2010][Beta][0.16.4] starting up...
EDMA Startup process... OK
(EDMA_DII)[DII_Request] Method: test_in_int signature S32U32S16U16S8U8rS32
(EDMA_DII)[DII_add_canonical_int] Param 1 Value: '-100000' Flags:0 Param Signature: S32
(EDMA_DII)[DII_add_canonical_int] Param 2 Value: '100000' Flags:0 Param Signature: U32
(EDMA_DII)[DII_add_canonical_int] Param 3 Value: '-10000' Flags:0 Param Signature: S16
(EDMA_DII)[DII_add_canonical_int] Param 4 Value: '10000' Flags:0 Param Signature: U16
(EDMA_DII)[DII_add_canonical_int] Param 5 Value: '-100' Flags:0 Param Signature: S8
(EDMA_DII)[DII_add_canonical_int] Param 6 Value: '100' Flags:0 Param Signature: U8
=================> Running Method
(EDMA_DII)[DII_Invoke] Invoking Method test_in_int#S32U32S16U16S8U8rS32
(DII_TEST)[DII_TESTtest_in_intS32U32S16U16S8U8rS32] Signed 32-bit parameter   : -100000
(DII_TEST)[DII_TESTtest_in_intS32U32S16U16S8U8rS32] Unsigned 32-bit parameter : 100000
(DII_TEST)[DII_TESTtest_in_intS32U32S16U16S8U8rS32] Signed 16-bit parameter   : -10000
(DII_TEST)[DII_TESTtest_in_intS32U32S16U16S8U8rS32] Unsigned 16-bit parameter : 10000
(DII_TEST)[DII_TESTtest_in_intS32U32S16U16S8U8rS32] Signed 8-bit parameter   : -100
(DII_TEST)[DII_TESTtest_in_intS32U32S16U16S8U8rS32] Unsigned 8-bit parameter : 100
=================> Method Finished =========================>
(EDMA_DII)[DII_Request] Method: test_in_real signature R32R64rS32
(EDMA_DII)[DII_add_canonical_double] Param 1 Value: 5.123000 Flags:-0.000000 Param Signature: (null)
(EDMA_DII)[DII_add_canonical_double] Param 2 Value: -500.321000 Flags:-0.000000 Param Signature: (null)
=================> Running Method
(EDMA_DII)[DII_Invoke] Invoking Method test_in_real#R32R64rS32
(DII_TEST)[DII_TESTtest_in_realR32R64rS32] Real 32-bit parameter : 5.123000
(DII_TEST)[DII_TESTtest_in_realR32R64rS32] Real 64-bit parameter : -500.321000
=================> Method Finished =========================>
(EDMA_DII)[DII_Request] Method: test_in_mixed signature S8U32R32S32S32
(EDMA_DII)[DII_add_canonical_int] Param 1 Value: '5' Flags:0 Param Signature: S8
(EDMA_DII)[DII_add_canonical_int] Param 2 Value: '10000' Flags:0 Param Signature: U32
(EDMA_DII)[DII_add_canonical_double] Param 3 Value: 100.523000 Flags:-0.000000 Param Signature: (null)
(EDMA_DII)[DII_add_canonical_int] Param 4 Value: '-10000' Flags:0 Param Signature: S32
=================> Running Method
(EDMA_DII)[DII_Invoke] Invoking Method test_in_mixed#S8U32R32S32S32
(DII_TEST)[DII_TESTtest_in_mixedS8U32R32S32S32] Signed 8-bit parameter   : 5
(DII_TEST)[DII_TESTtest_in_mixedS8U32R32S32S32] Unsigned 32-bit parameter : 10000
(DII_TEST)[DII_TESTtest_in_mixedS8U32R32S32S32] Real 32-bit parameter : 100.523003
(DII_TEST)[DII_TESTtest_in_mixedS8U32R32S32S32] Signed 32-bit parameter   : -10000
=================> Method Finished =========================>
(EDMA_DII)[DII_Request] Method: test_in_other signature ZOZrS32
(EDMA_DII)[DII_add_canonical_int] Param 1 Value: Hello World Flags:0 Param Signature: Z
(EDMA_DII)[DII_add_canonical_int] Param 2 Value: '0' Flags:0 Param Signature: O
(EDMA_DII)[DII_add_canonical_int] Param 3 Value: bye cruel World! Flags:0 Param Signature: Z
=================> Running Method
(EDMA_DII)[DII_Invoke] Invoking Method test_in_other#ZOZrS32
(DII_TEST)[DII_TESTtest_in_otherZOZrS32] First Input String is  : 'Hello World'
(DII_TEST)[DII_TESTtest_in_otherZOZrS32] Second Input String is : 'bye cruel World!'
(DII_TEST)[DII_TESTtest_in_otherZOZrS32] Object Input Param is  : 0 ==> Object Report follows
==OBJECT REPORT==================================================
Object (0) of class (DII_TEST)
Top Level Object. Object haven't father
No PseudoFather information
Object has 0 superobjects
Object has 0 subobjects
Virtual Methods in object
==END OBJECT REPORT==========================================
=================> Method Finished =========================>
(EDMA_DII)[DII_Request] Method: test_out_int signature sS8sU32sS32
(EDMA_DII)[DII_add_canonical_int] Param 1 Value: 0x0xbf8529fb Flags:0 Param Signature: sS8
(EDMA_DII)[DII_add_canonical_int] Param 2 Value: 0x0xbf8529e8 Flags:0 Param Signature: sU32
(EDMA_DII)[DII_add_canonical_int] Param 3 Value: 0x0xbf8529ec Flags:0 Param Signature: sS32
=================> Running Method
(EDMA_DII)[DII_Invoke] Invoking Method test_out_int#sS8sU32sS32
(DII_TEST)[DII_TESTtest_out_intsS8sU32sS32] Parameter pointer received: 0xbf8529fb - 0xbf8529e8 - 0xbf8529ec
(DII_TEST)[DII_TESTtest_out_intsS8sU32sS32] Input Signed 8-bit Integer is    : 100 
(DII_TEST)[DII_TESTtest_out_intsS8sU32sS32] Input UnSigned 32-bit Integer is : 10000 
(DII_TEST)[DII_TESTtest_out_intsS8sU32sS32] Input Signed 32-bit Integer is   : -10000 
=================> Method Finished =========================>
Output parameters are:
--> Signed 8-bit Integer    : 1 [0xbf8529fb]
--> Unsigned 32-bit Integer : 2 [0xbf8529e8]
--> Signed 32-bit Integer   : 3 [0xbf8529ec]
(EDMA_DII)[DII_Request] Method: test_out_real signature sR32sR64rS32
(EDMA_DII)[DII_add_canonical_int] Param 1 Value: 0x0xbf8529e4 Flags:0 Param Signature: sR32
(EDMA_DII)[DII_add_canonical_int] Param 2 Value: 0x0xbf8529d8 Flags:0 Param Signature: sR64
=================> Running Method
(EDMA_DII)[DII_Invoke] Invoking Method test_out_real#sR32sR64rS32
(DII_TEST)[DII_TESTtest_out_realsR32sR64rS32] Values received -> r32:10.560000 r64:501.760000
=================> Method Finished =========================>
Output parameters are:
--> Real 32-bit Integer : 20.560001 [0xbf8529fb]
--> Real 64-bit Integer : 511.760000 [0xbf8529e8]
(EDMA_DII)[DII_Request] Method: test_in_other signature ZOZrS32
(EDMA_DII)[DII_add_canonical_int] Param 1 Value: Hello Flags:0 Param Signature: Z
(EDMA_DII)[DII_add_canonical_int] Param 2 Value: '0' Flags:0 Param Signature: O
(EDMA_DII)[DII_add_canonical_int] Param 3 Value: World Flags:0 Param Signature: Z
=================> Running Method
(EDMA_DII)[DII_Invoke] Invoking Method test_in_other#ZOZrS32
(DII_TEST)[DII_TESTtest_in_otherZOZrS32] First Input String is  : 'Hello'
(DII_TEST)[DII_TESTtest_in_otherZOZrS32] Second Input String is : 'World'
(DII_TEST)[DII_TESTtest_in_otherZOZrS32] Object Input Param is  : 0 ==> Object Report follows
==OBJECT REPORT==================================================
Object (0) of class (DII_TEST)
Top Level Object. Object haven't father
No PseudoFather information
Object has 0 superobjects
Object has 0 subobjects
Virtual Methods in object
==END OBJECT REPORT==========================================
(EDMA_DII)[DII_Request] Method: test_out_other signature sZsOsZrS32
(EDMA_DII)[DII_add_canonical_int] Param 1 Value: 0x0xbf852a4c Flags:0 Param Signature: sZ
(EDMA_DII)[DII_add_canonical_int] Param 2 Value: 0x0xbf8529e0 Flags:0 Param Signature: sO
(EDMA_DII)[DII_add_canonical_int] Param 3 Value: 0x0xbf8529fc Flags:0 Param Signature: sZ
=================> Running Method
(EDMA_DII)[DII_Invoke] Invoking Method test_out_other#sZsOsZrS32
(DII_TEST)[DII_TESTtest_out_othersZsOsZrS32] First Input String is: 'Hello'
(DII_TEST)[DII_TESTtest_out_othersZsOsZrS32] Second Input String is : 'World!'
(DII_TEST)[DII_TESTtest_out_othersZsOsZrS32] Object Input Param is  : 1 ==> Object Report follows
==OBJECT REPORT==================================================
Object (1) of class (EDMA_DII)
Top Level Object. Object haven't father
No PseudoFather information
Object has 0 superobjects
Object has 0 subobjects
Virtual Methods in object
==END OBJECT REPORT==========================================
=================> Method Finished =========================>
Output parameters are:
---> String 1 : 'Parameter 1 Changed!' 
---> Object   : -1
---> String 2 : 'Parameter 2 Changed!'
(EDMA_DII)[DII_Request] Method: test_inout signature sR64ZsZS32sA
(EDMA_DII)[DII_add_canonical_int] Param 1 Value: 0x0xbf8529d8 Flags:0 Param Signature: sR64
(EDMA_DII)[DII_add_canonical_int] Param 2 Value: Hello Flags:0 Param Signature: Z
(EDMA_DII)[DII_add_canonical_int] Param 3 Value: 0x0xbf852a4c Flags:0 Param Signature: sZ
(EDMA_DII)[DII_add_canonical_int] Param 4 Value: '54' Flags:0 Param Signature: S32
(EDMA_DII)[DII_add_canonical_int] Param 5 Value: '100' Flags:0 Param Signature: S32
(EDMA_DII)ERROR: Type mismatch: Type S32 not valid for parameter 4 on method 'test_inoutsR64ZsZS32sA'.
=================> Running Method
(EDMA_DII)[DII_Invoke] Invoking Method test_inout#sR64ZsZS32sA
(DII_TEST)[DII_TESTtest_inoutsR64ZsZS32sA] Input Real 64-bits : 101.790000
(DII_TEST)[DII_TESTtest_inoutsR64ZsZS32sA] Input string       :'Hello'
(DII_TEST)[DII_TESTtest_inoutsR64ZsZS32sA] Input string       :'World!!!!'
(DII_TEST)[DII_TESTtest_inoutsR64ZsZS32sA] Input Int 32-bits  : 54
=================> Method Finished =========================>
Output parameters are:
---> String 1 : 'String parameter changed!' 
---> Real 64  : 121.790000
---------------------[EDMAEnd]----------------------------------
No more process in GNU EDMA. Freeing All Resources...
Cleanup done!!
